contracts/access/manager/AccessManager.sol

```diff
@@ -0,0 +1,703 @@
+     */
+    function setContractModeCustom(address target) public virtual onlyGroup(ADMIN_GROUP) {
+        // todo set delay or document risks
+        _setContractMode(target, AccessMode.Custom);
+    }
+
+    /**
+     * @dev Set the operating mode of a contract to Open. This allows anyone to call any `restricted()` function with
+     * no delay.
+     *
+     * Requirements:
+     *
+     * - the caller must be a global admin
+     *
+     * Emits a {AccessModeUpdated} event.
+     */
+    function setContractModeOpen(address target) public virtual onlyDelayedAdmin {
+        // todo set delay or document risks
+        _setContractMode(target, AccessMode.Open);
+    }
+
+    /**
+     * @dev Set the operating mode of a contract to Close. This prevents anyone from calling any `restricted()`
+     * function.
+     *
+     * Requirements:
+     *
+     * - the caller must be a global admin
+     *
+     * Emits a {AccessModeUpdated} event.
+     */
+    function setContractModeClosed(address target) public virtual onlyGroup(ADMIN_GROUP) {
+        // todo set delay or document risks
+        _setContractMode(target, AccessMode.Closed);
+    }
+
+    /**
+     * @dev Set the operating mode of a contract. This is an internal setter with no access restrictions.
+     *
+     * Emits a {AccessModeUpdated} event.
+     */
+    function _setContractMode(address target, AccessMode mode) internal virtual {
+        _contractMode[target] = mode;
+        emit AccessModeUpdated(target, mode);
+    }
+
+    // ============================================== DELAYED OPERATIONS ==============================================
+    /**
+     * @dev Return the timepoint at which a scheduled operation will be ready for execution. This returns 0 if the
+     * operation is not yet scheduled, was executed or was canceled.
+     */
+    function getSchedule(bytes32 id) public view virtual returns (uint48) {
+        return _schedules[id];
+    }
+
+    /**
+     * @dev Schedule a delayed operation for future execution, and return the operation identifier. It is possible to
+     * choose the timestamp at which the operation becomes executable as long as it satisfies the execution delays
+     * required for the caller. The special value zero will automatically set the earliest possible time.
+     *
+     * Emits a {Scheduled} event.
+     */
+    function schedule(address target, bytes calldata data, uint48 when) public virtual returns (bytes32) {
+        address caller = _msgSender();
+        bytes4 selector = bytes4(data[0:4]);
+
+        // Fetch restriction to that apply to the caller on the targeted function
+        (bool allowed, uint32 setback) = canCall(caller, target, selector);
+
+        uint48 minWhen = Time.timestamp() + setback;
+
+        // If caller is not authorised, revert
+        if (!allowed && (setback == 0 || when != 0 && when < minWhen)) {
+            revert AccessManagerUnauthorizedCall(caller, target, selector);
+        }
```

Comment:
I think we don't want to offer error interface stability. I know @frangio has a preference for fewer errors and I'd second that because I see value in trying to minimize the error offering.

Reply:
OK, but is scheduling something too early realy a "you are not authorized to call" error ?

Reply:
I'd say the authorization comes from the schedule as well, so any action taken by an allowed user but out of time should be considered "Unauthorized" imo